### PR TITLE
Adding rbd image tests on ecpool

### PR DESCRIPTION
**Pull request by manasagowri**
Signed-off-by: mgowri <mgowri@redhat.com>

Test Run : https://jenkins.ceph.redhat.com/view/RHCS%20QE/job/rhceph-test-executor/1978/console
Success Log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-DMOM10/test_image_on_ecpool_0.log

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/rbd/rbd_ecpool_test_image.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        int: The return value. 0 for success, 1 otherwise
+    """
+    rbd = Rbd(**kw)
+    pool = kw["config"].get("pool", rbd.random_string())
+    image = kw["config"].get("image", rbd.random_string())
+    size = kw["config"].get("image_size", "10G")
+
+    try:
+        if not rbd.create_pool(poolname=pool):
+            # create pool does not catch exceptions, it returns true/false.
```
Comment (rkhambad) [inline tests/rbd/rbd_ecpool_test_image.py:31]: we can use initial_rbd_config method once PR https://github.com/red-hat-storage/cephci/pull/1991 is merged.

File: suites/pacific/rbd/tier-1_rbd.yaml [hunk 2]
```diff
@@ -19,6 +19,8 @@ tests:
       abort-on-fail: true
       module: install_prereq.py
       name: install ceph pre-requisites
+      config:
```
Comment (rkhambad) [inline suites/pacific/rbd/tier-1_rbd.yaml:22]: is this required to add is_production : True?

File: utility/utils.py [hunk 3]
```diff
@@ -1446,6 +1446,9 @@ def run_fio(**fio_args):
     else:
         opt_args = f"--ioengine=rbd --rbdname={fio_args['image_name']} --pool={fio_args['pool_name']}"
 
+    if fio_args.get("size"):
+        opt_args += f" --size={fio_args.get('size')}"
+
```
Comment (VasishtaShastry) [inline utility/utils.py:1451]: Can you please check line 1441
Can we please optimise usage of size in this method ?

File: suites/pacific/rbd/tier-1_rbd.yaml (hunk 2 above)
Comment (VasishtaShastry) [inline suites/pacific/rbd/tier-1_rbd.yaml:22]: Can we please review is_production usage here ?

**Review by VasishtaShastry — CHANGES_REQUESTED** [review]

File: utility/utils.py (hunk 3 above)
Comment (manasagowri) [reply utility/utils.py:1451]: @VasishtaShastry But in my case I am not passing any filename all i want to specify is the size of IOs to be written, how do you think we can optimise this? is file_size a required option when file_name is specified or can it be optional?

File: tests/rbd/rbd_ecpool_test_image.py (hunk 1 above)
Comment (manasagowri) [reply tests/rbd/rbd_ecpool_test_image.py]: Done.

File: suites/pacific/rbd/tier-1_rbd.yaml (hunk 2 above)
Comment (manasagowri) [reply suites/pacific/rbd/tier-1_rbd.yaml]: Removed

Comment (manasagowri) [reply suites/pacific/rbd/tier-1_rbd.yaml]: Removed

**Comment by manasagowri** [comment]
Latest logs - https://jenkins.ceph.redhat.com/view/RHCS%20QE/job/rhceph-test-executor/2003/console

File: tests/rbd/rbd_utils.py [hunk 4]
```diff
@@ -344,3 +362,97 @@ def clean_up(self, **kw):
...
+               ec_pool_config:
+                  pool: rbd_pool_4
+                  data_pool: rbd_ec_pool_4
+                  ec_profile: rbd_ec_profile_4
+                  image: rbd_image_4
+                  size: 10G
+               rep_pool_config:
+                  pool: rbd_rep_pool
+                  image: rbd_rep_image
+                  size: 10G
```
Comment (VasishtaShastry) [inline tests/rbd/rbd_utils.py:384]: [Optional]
Does it makes sense to add info on defaults ?

File: utility/utils.py (hunk 3 above)
Comment (VasishtaShastry) [reply utility/utils.py:1451]: @manasagowri I think its required, I'm not sure for now
But It has been defaulted to 100M for now

Why can't we just remove `--size={fio_args.get('file_size', '100M')}` from 1441 and update method doc
(And see if anywhere in yamls or methods we are usig file_size and change it to size)

File: tests/rbd/rbd_utils.py [hunk 5]
```diff
@@ -344,3 +362,97 @@ def clean_up(self, **kw):
...
+    """
```
[the remark this answers is not in the document]
Comment (manasagowri) [reply tests/rbd/rbd_utils.py:385]: Done.

**Comment by manasagowri** [comment]
@VasishtaShastry done changes with respect to size as well. Updated the one file which was using file_size parameter to use size instead.

**Review by VasishtaShastry — APPROVED** [review]

**Review by Manohar-Murthy — APPROVED** [review]
Approving based on the group review.

**Pull request merged**